src/Microsoft.CodeAnalysis.Analyzers/CSharp/MetaAnalyzers/Fixers/CSharpPreferIsKindFix.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.Fixers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpPreferIsKindFix))]
+    [Shared]
+    public sealed class CSharpPreferIsKindFix : PreferIsKindFix
+    {
+        protected override async Task<Document> ConvertKindToIsKindAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var binaryExpression = root.FindNode(sourceSpan, getInnermostNodeForTie: true).FirstAncestorOrSelf<BinaryExpressionSyntax>();
+
+            if (!(binaryExpression.Left is InvocationExpressionSyntax invocation))
```

Comment:
```suggestion
            if (!(binaryExpression?.Left is InvocationExpressionSyntax invocation))
```

Reply:
Unless some other analyzer author writes an analyzer with same ID that reports diagnostic at different location?

Reply:
IMO, i think it's valid for diagnostic IDEs to claim squatting rights.  other libraries should not produce these IDs, and fixers should not have to be resilient to someone trying to spoof the diagnsotic.

this does't hold for compiler errors of course, as they're free to change those will nilly.  but for bound analyzer/fixer libs, i would prefer that philosophy myself.

Reply:
Even if it doesn't hold, you'd just get a gold bar and everything else keeps working.